source/enemyLevel.js

```diff
@@ -0,0 +1,133 @@
+        this.load.image('red-hitsplat', 'source/assets/RedHitsplat.png');
+    }
+
+    childCreate() {
+        // Create kill quest
+        this.killQuestText = this.add.text(530, 270, "", {fill: 'white'}).setDepth(3);
+        this.clickObjectMetaData.forEach((enemy, index) => {
+            this.killQuestText.text += this.characterData[this.currentLevel].enemiesKilled[enemy.name] + "/" + this.killQuest + " " + enemy.name + "s";
+            if (index + 1 < this.clickObjectMetaData.length) {
+                this.killQuestText.text += " & ";
+            }
+        });
+
+        this.questCompleteText = this.add.text(530, 290, 'Quest complete!', {fill: 'white'}).setDepth(3);
+
+        // Hide text if level quest has not been completed
+        if (!this.characterData[this.currentLevel].questCompleted) {
+            this.questCompleteText.visible = false;
+        }
+
+        // Button text to test autoclickers
+        let autoClickerButton = this.add.text(530, 250, '50 gold for autoclicker', {fill: 'white'}).setDepth(3);
+        autoClickerButton.setInteractive();
+        autoClickerButton.on("pointerup", ()=>{
+            if (this.characterData.gold >= 50) {
+                this.addGold(-50);
+                this.createAutoClicker({dps: 5, level: 1, type: 'Hired Bowman'});
+            }           
+        });
+
+        // Create click objects
+        this.clickObjectMetaData.forEach((clickObject) => {
+            this.clickObjects.push(
+                new Enemy({
+                    scene: this,
+                    x: this.width/2-100,
+                    y: this.height/2-115,
+                    maxHealth: clickObject.maxHealth,
+                    name: clickObject.name,
+                    killGold: clickObject.killGold
+                })
+            );
+        });
+
+        // Show stats
+        let statColor = 'white';
+        this.autoClickDpsText = this.add.text(20, 120, "AutoClicker DPS: " + this.autoClickDps, {fill: statColor}).setDepth(3);
+
+        // Re-add autoclickers from cookies on first load
+        if (this.characterData.hasCookies && this.autoClickers.length == 0){
+            let numAutoClickers = this.characterData.numberOfAutoClickers;
+            this.characterData.numberOfAutoClickers = 0;
+            this.autoClickDps = 0;
+            this.updateAutoClickerDPS(0);
+            for (let i = 0; i < numAutoClickers; i++) {
+                this.createAutoClicker({dps: 5, level: 1, type: 'Hired Bowman'});
+            }
+        }
+    }
+
+    enemyKilled(name){
+        this.characterData.totalEnemiesKilled++;
+        // Update kill quest score
+        if (this.characterData[this.currentLevel].enemiesKilled[name] < this.killQuest) {
+            this.characterData[this.currentLevel].enemiesKilled[name]++;
+
+            let questCompleted = true;
+            this.killQuestText.text = "";
+            this.clickObjectMetaData.forEach((enemy, index) => {
```

Comment:
1) They're the same, I mustve messed up copying it in there. 
2) Yeah I'm trying to think of a way I can abstract this but still keep the enemy related stuff in the enemyLevel class. Suggestions are welcome. I was mainly doing this because I thought that once we start actually getting resources from the trees and whatever stuff we do then we'd have to add more into this file. But I don't want to have a bunch of duplicate code

Reply:
I didn't completely fix the duplicate code but I reduced it some by moving that one method back into Level.js and deleting ResourceLevel.js since it was tiny and functionally useless. I think later when resources are hashed out better it will be easier to clean this up fully.